OS/DiskOS/Programs/Disks/load.lua

```diff
@@ -1,39 +1,54 @@
+-- Load a game into memory either from a .png or .lk12 file or from the clipboard
+
 local source = select(1,...)
 
 local term = require("terminal")
 local eapi = require("Editors")
 
 local png = false
 
-if source and source ~= "@clip" and source ~= "-?" then
-  source = term.resolve(source)
-  if source:sub(-4,-1) == ".png" then
-    png = true
-  elseif source:sub(-5,-1) ~= ".lk12" then
-    local lksrc = source..".lk12"
-    if fs.exists(lksrc) then
-      source = lksrc
-    elseif fs.exists(source..".png") then
-      source = source..".png"
+local lk12Data = nil
+
+-- Load from clipboard
+if source == "@clip" then
+  lk12Data = clipboard()
+else
+  -- Use the last know path if no file was provided
+  if not source then
+    source = eapi.filePath
+  -- Load from the provided file
+  elseif source ~= "-?" then
+    -- Check if the file exists and is not a directory
+    source = term.resolve(source)
+    if not fs.exists(source) then return 1, "File "..source.." does not exist" end
+    if fs.isDirectory(source) then return 1, "Could not load "..source.." as it is a directory" end
+    -- Load differently according to the type of the file
+    if source:sub(-4,-1) == ".png" then
       png = true
-    else
-      source = lksrc
+    elseif source:sub(-5,-1) ~= ".lk12" then
```

Comment:
The `.lk12` insertion may be too late, because the program my be already quit with the message `File ... does not exist`